stetho/src/main/java/com/facebook/stetho/inspector/protocol/module/CSS.java

```diff
@@ -98,75 +92,114 @@ public void store(String name, String value) {
+
+            final CSSRule rule = new CSSRule();
+            rule.origin = Origin.REGULAR;
+            rule.selectorList = new SelectorList();
+            rule.selectorList.selectors = ListUtil.newImmutableList(selector);
+            rule.style = new CSSStyle();
+            rule.style.cssProperties = properties;
+            rule.style.shorthandEntries = Collections.emptyList();
+
+            if (editable) {
+              rule.style.styleSheetId = String.format("%d.%s", request.nodeId, selector.value);
+            }
+
+            mDocument.getElementStyles(elementForNodeId, ruleName, new StyleAccumulator() {
               @Override
-              public void store(String selector, String name, String value, boolean isDefault) {
-                if (!isDefault) {
-                  ArrayList<CSSProperty> properties = mSelectorCollector.get(selector);
-                  if (properties == null) {
-                    properties = new ArrayList<>();
-                    mSelectorCollector.put(selector, properties);
-                  }
-
-                  final CSSProperty property = new CSSProperty();
-                  property.name = name;
-                  property.value = value;
-                  properties.add(property);
-                }
+              public void store(String name, String value, boolean isDefault) {
+                final CSSProperty property = new CSSProperty();
+                property.name = name;
+                property.value = value;
+                properties.add(property);
               }
             });
+
+            match.rule = rule;
+            result.matchedCSSRules.add(match);
+          }
+        });
       }
     });
 
-    final GetMatchedStylesForNodeResult result = new GetMatchedStylesForNodeResult();
-    result.matchedCSSRules = new ArrayList<>();
-    result.inherited = Collections.emptyList();
-    result.pseudoElements = Collections.emptyList();
+    return result;
+  }
 
-    for (Map.Entry<String, ArrayList<CSSProperty>> entry : mSelectorCollector.entrySet()) {
-      final ArrayList<CSSProperty> properties = entry.getValue();
-      if (properties.isEmpty()) {
-        continue;
-      }
+  @ChromeDevtoolsMethod
+  public SetPropertyTextResult setPropertyText(JsonRpcPeer peer, JSONObject params) {
+    final SetPropertyTextRequest request =  mObjectMapper.convertValue(
+            params,
+            SetPropertyTextRequest.class);
 
-      final RuleMatch match = new RuleMatch();
-      match.matchingSelectors = ListUtil.newImmutableList(0);
+    final String[] parts = request.styleSheetId.split("\\.", 2);
+    final int nodeId = Integer.parseInt(parts[0]);
+    final String ruleName = parts[1];
 
-      final Selector selector = new Selector();
-      selector.value = entry.getKey();
+    final String[] keyValue = request.text.split(":", 2);
+    final String key = keyValue[0].trim();
+    final String value = keyValue[1].replace(";", "").trim();
```

Comment:
Every time you call `String#replace(String, String)`, it compiles a regular expression via `Pattern`. This can be quite expensive! 

For this super simple replacement, let's put a utility method over into `java.com.facebook.stetho.common.StringUtil` :

```
public static String remove(String string, char target) {
  int length = string.length();
  StringBuilder builder = new StringBuilder(length);
  for (int i = 0; i < length; ++i) {
    char c = string.charAt(i);
    if (c != target) {
      builder.append(c);
    }
  }
  return builder.toString();
}
```

(note that `String#split(String, int)` also interprets as a regex, but if you look at the implementation it has a fast path that covers exactly how you're using it here)